_search-plugins/retrieve-specific-fields.md

```diff
@@ -0,0 +1,877 @@
+
+## Specifying the fields to retrieve
+
+You can list the fields of interest in the search request using `fields` parameter. Wildcard patterns are also accepted. See following example:
+
+```
+GET "/index1/_search?pretty"
+{
+    "_source": false,
+    "fields": ["age", "nam*"],
+    "query": {
+        "match_all": {}
+  }
+}
+```
+
+Example response:
+
+```
+{
+  "hits" : {
+    "total" : {
+      "value" : 2,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "index1",
+        "_id" : "41",
+        "_score" : 1.0,
+        "fields" : {
+          "name" : [
+            "John Doe"
+          ],
+          "age" : [
+            30
+          ]
+        }
+      },
+      {
+        "_index" : "index1",
+        "_id" : "51",
+        "_score" : 1.0,
+        "fields" : {
+          "name" : [
+            "Jane Smith"
+          ],
+          "age" : [
+            25
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+### Extracting fields with custom format
+
+You can also use object notation, to apply a custom format to the chosen field, see following example.
+
+If you have the following document:
+
+```
+{
+  "_index": "my_index",
+  "_type": "_doc",
+  "_id": "1",
+  "_source": {
+    "title": "Document 1",
+    "date": "2023-07-04T12:34:56Z"
+  }
+}
+```
+
+You can query with `fields` parameter and custom format:
+
+```
+GET /my_index/_search
+{
+  "query": {
+    "match_all": {}
+  },
+  "fields": {
+    "date": {
+      "format": "yyyy-MM-dd"
+    }
+  },
+  "_source": false
+}
+```
+
+Additionally, you can also use [most fields]({{site.url}}{{site.baseurl}}/query-dsl/full-text/multi-match/#most-fields) and [field aliases]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/alias/) with `fields` parameter, as it queries both document `_source` and `_mappings` of the index.
+
+## Searching with `docvalue_fields`
+
+`docvalue_fields` is another parameter you can use in OpenSearch to retrieve specific fields from the index, but it works slightly differently compared to the `fields` parameter. The `docvalue_fields` parameter retrieves details from doc values rather than from the `_source` field, which is more efficient for fields that are not analyzed, like keyword, date, and numeric fields. Doc values are a columnar storage format optimized for efficient sorting and aggregations. It stores the values on disk in a way that is easy to read. When you use `docvalue_fields`, OpenSearch reads the values directly from this optimized storage format. It is useful for retrieving values of fields that are primarily used for sorting, aggregations, and for use in scripts.
```

Comment:
```suggestion
To retrieve specific fields from the index, you can also use the `docvalue_fields` parameter. This parameter works slightly differently compared to the `fields` parameter. It retrieves details from doc values rather than from the `_source` field, which is more efficient for fields that are not analyzed, like keyword, date, and numeric fields. Doc values have a columnar storage format optimized for efficient sorting and aggregations. It stores the values on disk in a way that is easy to read. When you use `docvalue_fields`, OpenSearch reads the values directly from this optimized storage format. It is useful for retrieving values of fields that are primarily used for sorting, aggregations, and for use in scripts.
```